paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala

```diff
@@ -0,0 +1,125 @@
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.{CatalystTypeConverters, InternalRow}
+import org.apache.spark.sql.connector.catalog.SupportsPartitionManagement
+import org.apache.spark.sql.types.StructType
+
+import java.util
+import java.util.{Collections, UUID}
+
+import scala.collection.JavaConverters._
+
+abstract class PaimonPartitionManagement(table: Table) extends SupportsPartitionManagement {
+
+  override def partitionSchema(): StructType = {
+    val partitionKeys = table.partitionKeys
+    val rowType = new RowType(
+      table.rowType.getFields.asScala
+        .filter(dataFiled => partitionKeys.contains(dataFiled.name()))
+        .asJava)
+    SparkTypeUtils.fromPaimonRowType(rowType)
+  }
+
+  override def dropPartition(internalRow: InternalRow): Boolean = {
+    val partitionKeys = table.partitionKeys
+    val rowType = new RowType(
+      table.rowType.getFields.asScala
+        .filter(dataFiled => partitionKeys.contains(dataFiled.name()))
+        .asJava)
+    // convert internalRow to row
+    val row: Row = CatalystTypeConverters
+      .createToScalaConverter(partitionSchema())
+      .apply(internalRow)
+      .asInstanceOf[Row]
+    val rowDataPartitionComputer = new RowDataPartitionComputer(
+      FileStorePathFactory.PARTITION_DEFAULT_NAME.defaultValue,
+      rowType,
+      partitionKeys.asScala.toArray)
+    val partitionMap = rowDataPartitionComputer.generatePartValues(new SparkRow(rowType, row))
+    val identifier: Long = BatchWriteBuilder.COMMIT_IDENTIFIER
+    val commit: FileStoreCommit =
+      (table.asInstanceOf[AbstractFileStoreTable]).store.newCommit(UUID.randomUUID.toString)
+    commit.dropPartitions(Collections.singletonList(partitionMap), identifier)
+    true
+  }
+
+  override def replacePartitionMetadata(
+      ident: InternalRow,
+      properties: util.Map[String, String]): Unit = {
+    throw new UnsupportedOperationException("Replace partition is not supported")
+  }
+
+  override def loadPartitionMetadata(ident: InternalRow): util.Map[String, String] = {
+    throw new UnsupportedOperationException("Load partition is not supported")
+  }
+
+  override def listPartitionIdentifiers(
+      partitionCols: Array[String],
+      internalRow: InternalRow): Array[InternalRow] = {
+    assert(
+      partitionCols.length == internalRow.numFields,
+      s"Number of partition names (${partitionCols.length}) must be equal to " +
+        s"the number of partition values (${internalRow.numFields})."
+    )
+    val schema: StructType = partitionSchema
+    assert(
+      partitionCols.forall(fieldName => schema.fieldNames.contains(fieldName)),
+      s"Some partition names ${partitionCols.mkString("[", ", ", "]")} don't belong to " +
+        s"the partition schema '${schema.sql}'."
+    )
+
+    val tableScan: TableScan = table.newReadBuilder.newScan
+    val binaryRows: List[BinaryRow] = tableScan.listPartitions.asScala.toList
```

Comment:
the current implement is ok. but maybe an optimization can be considered, that is we can convert `partitionCols` and `internalRow` to the Paimon `Predicate`, and `newReadBuilder.withFilter(...).newScan` to get the result directly. That's just a suggestion, you can stay this way.

Reply:
```
      val predicate = PredicateBuilder.partition(partitionMap, tableRowType)
      getTable.newReadBuilder.withFilter(predicate).newScan()
      val binaryRows = tableScan.listPartitions.asScala.toList
```
I tried using partition filters before, but tableScan. listPartitions is empty.